pxr/base/arch/fileSystem.cpp

```diff
@@ -479,6 +491,48 @@ ArchGetFileName(FILE *file)
         result.assign(buf, buf + r);
     }
     return result;
+#elif defined (ARCH_OS_FREEBSD)
+    int mib[4];
```

Comment:
FYI: FreeBSD 14 will be having F_KINFO fcntl flag https://github.com/freebsd/freebsd-src/blob/97722455cce4f8398f6411eaaa80d595dcb5fb3c/lib/libc/sys/fcntl.2#L197

already present in FreeBSD current (and functioning well), most likely noticeably faster that going through file descriptors (but no choice).